### PR TITLE
Update create template tutorial for .NET 6

### LINKED ISSUE #27977: Add a sample link of multiple templates within one Nuget package
I've packed a Nuget with my project similar to the following one of doc.

```
project_folder
│   MyDotnetTemplates.csproj
│
└───templates
    ├───mytemplate1
    │   │   console.cs
    │   │   readme.txt
    │   │
    │   └───.template.config
    │           template.json
    │
    └───mytemplate2
        │   otherfile.cs
        │
        └───.template.config
                template.json
```

But when I use `dotnet new` command, the CLI informs me of conflicted templates. Will it be better if there is a repo sample link here?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4e7deb06-37bb-b659-7cf3-9c47a91e065e
* Version Independent ID: d6e6f989-3956-6a0e-5e9c-fb3d1f9bd751
* Content: [Custom templates for dotnet new - .NET CLI](https://docs.microsoft.com/en-us/dotnet/core/tools/custom-templates#packing-a-template-into-a-nuget-package-nupkg-file)
* Content Source: [docs/core/tools/custom-templates.md](https://github.com/dotnet/docs/blob/main/docs/core/tools/custom-templates.md)
* Product: **dotnet-fundamentals**
* Technology: **dotnet-cli**
* GitHub Login: @adegeo
* Microsoft Alias: **tdykstra**

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Updated wording and content to account for .NET 6's new default console template project.
- Removed the editing of the project file step in the tutorial, as these were specific to .NET 5.
- Added a link to the samples repo. Dependent on dotnet/samples#4962

Fixes #27977

@tdykstra tagging you since you've worked in this area

### CODE REVIEW

File: docs/core/tutorials/cli-templates-create-project-template.md [hunk 1]
```diff
@@ -29,7 +31,7 @@ In this part of the series you'll learn how to:
 
 ## Create a project template
 
-Project templates produce ready-to-run projects that make it easy for users to start with a working set of code. .NET includes a few project templates such as a console application or a class library. In this example, you'll create a new console project that enables C# 10.0 and produces an `async main` entry point.
+Project templates produce ready-to-run projects that make it easy for users to start with a working set of code. .NET includes a few project templates such as a console application or a class library. In this example, you'll create a new console project that replaces the standard "Hello World" console output, with one that runs asynchronously.
```
Comment (CamSoper) [inline docs/core/tutorials/cli-templates-create-project-template.md:34]:
```suggestion
Project templates produce ready-to-run projects that make it easy for users to start with a working set of code. .NET includes a few project templates such as a console application or a class library. In this example, you'll create a new console project that replaces the standard "Hello World" console output with one that runs asynchronously.
```

**Review by CamSoper — DISMISSED** [review]

File: docs/core/tutorials/cli-templates-create-project-template.md [hunk 2]
```diff
@@ -43,28 +45,10 @@ working
 
 ## Modify Program.cs
 
-Open up the _program.cs_ file. The console project doesn't use an asynchronous entry point, so let's add that. Change your code to the following and save the file.
+Open up the _program.cs_ file. The standard console project doesn't asynchronously write to the console output, so let's add that. Change your code to the following and save the file:
```
Comment (tdykstra) [inline docs/core/tutorials/cli-templates-create-project-template.md:48]:
```suggestion
Open up the _program.cs_ file. The standard console project doesn't asynchronously write to the console output, so let's add that. Change the code to the following and save the file:
```

File: docs/core/tutorials/cli-templates-create-project-template.md [hunk 3]
```diff
@@ -139,7 +123,7 @@ The following template packages will be installed:
 Success: <root path>\working\templates\consoleasync installed the following templates:
 Templates                                         Short Name               Language          Tags
 --------------------------------------------      -------------------      ------------      ----------------------
-Example templates: async project                  consoleasync             [C#]              Common/Console/C#9
+Example templates: async project                  consoleasync             [C#]              Common/Console
```
Comment (tdykstra) [inline docs/core/tutorials/cli-templates-create-project-template.md:126]: above in line 111 
extra files folders and files you don't want
-->
extra folders and files you don't want

**Review by tdykstra — APPROVED** [review]
Looks good.  A couple of nit suggestions to change preexisting text.

**Pull request merged**